### PR TITLE
IPC JSON RPC (for external interface)

### PR DESCRIPTION
**Pull request by NikVolf**
nano-worker over `jsonrpc_core::IoHandler`

### CODE REVIEW

File: ipc/nano/src/lib.rs [hunk 1]
```diff
@@ -199,14 +202,136 @@ impl<S> Worker<S> where S: IpcInterface<S> {
...
+
+	/// IPC Server starts (non-blocking, in seprate thread)
+	pub fn start(&self) -> Result<(), SocketError> {
+		let mut worker = try!(IoHandlerWorker::new(&self.handler, &self.socket_addr));
+		self.is_stopping.store(false, Ordering::Relaxed);
+		let worker_is_stopping = self.is_stopping.clone();
+
+		::std::thread::spawn(move || {
+			while !worker_is_stopping.load(Ordering::Relaxed) {
+				worker.poll()
```
Comment (tomusdrw) [inline ipc/nano/src/lib.rs]: Would be good to handle possible panics.

File: ipc/nano/src/lib.rs [hunk 2]
```diff
@@ -199,14 +202,136 @@ impl<S> Worker<S> where S: IpcInterface<S> {
...
+		let fd = request.get_fds()[0]; 	// guaranteed to exist and be the only one
+										// because contains only immutable socket field as a member
+		if !fd.can_read() {
+			return;
+		}
+
+		unsafe { self.buf.set_len(0); }
+		match self.socket.nb_read_to_end(&mut self.buf) {
+			Ok(size) => {
+				if size == 0 {
```
Comment (tomusdrw) [inline ipc/nano/src/lib.rs]: `Ok(size) if size == 0 =>` ?

File: ipc/nano/src/lib.rs [hunk 3]
```diff
@@ -199,14 +202,136 @@ impl<S> Worker<S> where S: IpcInterface<S> {
...
+			while !worker_is_stopping.load(Ordering::Relaxed) {
+				worker.poll()
+			}
+		});
+
+		Ok(())
+	}
+
+	/// IPC server will eventually stop
+	pub fn stop(&self) {
```
Comment (tomusdrw) [inline ipc/nano/src/lib.rs]: How about implementing `Drop` and stopping the server when reference is dropped?
Or `start()` could return listening handle that when dropped would stop worker.

Comment (NikVolf) [inline ipc/nano/src/lib.rs]: Right

**Comment by gavofyork** [comment]
don't quite see what this achieves...

**Pull request merged**